upup/pkg/fi/cloudup/azure/azure_cloud.go

```diff
@@ -128,9 +134,34 @@ func (c *azureCloudImplementation) AddClusterTags(tags map[string]*string) {
 }
 
 func (c *azureCloudImplementation) GetApiIngressStatus(cluster *kops.Cluster) ([]kops.ApiIngressStatus, error) {
-	// TODO(kenji): Implement this. Currently we return nil as we
-	// don't create any resources for ingress to the API server.
-	return nil, nil
+	var ingresses []kops.ApiIngressStatus
+
+	// Get scale sets in cluster resource group and find masters scale set
+	scaleSets, err := c.vmscaleSetsClient.List(context.TODO(), cluster.Spec.CloudConfig.Azure.ResourceGroupName)
```

Comment:
How about ` cluster.AzureResourceGroupName()`? The function handles a case where a resource group is unspecified in the cluster spec (and kops creates a new resource group with the same name as the cluster).